packages/tools/kadena-cli/src/devnet/utils/simulation/coin/simulate.ts

```diff
@@ -0,0 +1,284 @@
+  tokenPool,
+  logFolder,
+  defaultChain,
+  seed,
+}: ISimulationOptions): Promise<void> {
+  const accounts: IAccount[] = [];
+
+  // Parameters validation
+  if (tokenPool < maxAmount) {
+    console.error(
+      'The max transfer amount cant be greater than the total token pool',
+    );
+    return;
+  }
+
+  if (numberOfAccounts <= 1) {
+    console.error('Number of accounts must be greater than 1');
+    return;
+  }
+
+  console.log('Seed value: ', seed);
+  const filepath = createFile(logFolder, `coin-${Date.now()}-${seed}.csv`);
+
+  try {
+    // Create accounts
+    for (let i = 0; i < numberOfAccounts; i++) {
+      // This will determine if the account has 1 or 2 keys (even = 1 key, odd = 2 keys)
+      const noOfKeys = i % 2 === 0 ? 1 : 2;
+      let account = await generateAccount(noOfKeys, defaultChain, network.host);
+      console.log(
+        `Generated KeyPair\nAccount: ${
+          account.account
+        }\nPublic Key: ${stringifyProperty(
+          account.keys,
+          'publicKey',
+        )}\nSecret Key: ${stringifyProperty(account.keys, 'secretKey')}\n`,
+      );
+
+      /* To diversify the initial testing sample, we cycle through all transfer types for the first funding transfers.
+      Subsequent transfers will be of the 'transfer' type to simulate normal operations. */
+      const fundingType =
+        i < simulationTransferOptions.length
+          ? simulationTransferOptions[i]
+          : 'transfer';
+
+      let result;
+      const sender: IAccount = { ...defaultAccount, chainId: '0' };
+
+      if (fundingType === 'cross-chain-transfer') {
+        account = {
+          ...account,
+          chainId: '1',
+        };
+
+        result = await crossChainTransfer({
+          network,
+          sender,
+          receiver: account,
+          amount: tokenPool / numberOfAccounts,
+          gasPayer: defaultAccount,
+        });
+      } else if (fundingType === 'safe-transfer') {
+        result = await safeTransfer({
+          network,
+          chainId: defaultChain,
+          receiver: account,
+          amount: tokenPool / numberOfAccounts,
+          sender,
+        });
+      } else {
+        result = await transfer({
+          network,
+          receiver: account,
+          chainId: defaultChain,
+          amount: tokenPool / numberOfAccounts,
+          sender,
+        });
+      }
+
+      // If the account is not in the accountlist, add it
+      if (accounts.includes(account)) {
+        throw Error('Duplicate account');
+      }
+      accounts.push(account);
+
+      await appendToLogFile(filepath, {
+        timestamp: Date.now(),
+        from: defaultAccount.account,
+        to: account.account,
+        amount: tokenPool / numberOfAccounts,
+        requestKey: result.reqKey,
+        action: 'fund',
+      });
+    }
+
+    // Generate first seeded random number
+    let seededRandomNo = seedRandom(seed);
+    let counter: number = 0;
+
+    while (true) {
```

Comment:
There is a lot of complexity within this while loop, making it hard for me to see how exactly it will break the loop, or if it is possible that it will run forever. This code would really benefit from moving some logic to separate methods

Reply:
Everything inside the loop is a little bit hard to refactor since I would have to pass a bunch of common variables inside each method and return objects with multiple properties due to the necessary logging in addition to some conditions that validate if the iteration is valid or if it should be skipped.

I can add some more comments to add clarity to what is happening and check if I can refactor it into smaller methods. 
Answering your question: the loop could go for infinity. This loop's goal is the simulate random traffic in the devnet (with the possibility of replicating it via the seed number). The only way to break the loop automatically is with a maximum time flag which is being developed on another PR, otherwise you'd have to shut it yourself.